enclave/core/calls.c

```diff
@@ -246,6 +247,51 @@ static oe_result_t _handle_call_enclave(uint64_t arg_in)
     return result;
 }
 
+/**
+ * The _oe_ecalls_table table is expected to be generated by oeedger8r
+ */
+extern const oe_ecall_func_t _oe_ecalls_table[];
+extern const size_t _oe_ecalls_table_size;
+
+/**
+ * This is the preferred way to call enclave functions.
+ */
+static oe_result_t _handle_call_enclave_function(uint64_t arg_in)
```

Comment:
why not make arg_in of type oe_call_enclave_args_t*?

Reply:
That is a fair point. I am using uint64_t for consistency with the other _handle functions nearby.
We could change all of them to take in typed pointers; but in that case the cast has to happen in the caller _handle_ecall. One could reason that it would be better to localize the cast/type information to the appropriate handle functions as opposed to the dispatching _handle_ecall function.

In any case, we can address the clean up separately in another PR if needed.

Reply:
This is done this way for consistency with other handle functions.